src/java/org/apache/cassandra/index/sai/disk/vector/OnDiskOrdinalsMap.java

```diff
@@ -259,6 +270,26 @@ public int getOrdinalForRowId(int rowId) throws IOException
             return reader.readInt();
         }
 
+        @Override
+        public void forEachOrdinalInRange(int startRowId, int endRowId, OrdinalConsumer consumer) throws IOException
+        {
+            reader.seek(rowOrdinalOffset);
```

Comment:
I think we need to seek to the index of `startRowId`

Reply:
Oops, commented before I understood the rest of the code. Is there a point where it'd make sense to seek to the starting row id index?

Reply:
maybe in some very extreme cases, ordinals map in megabytes and the values in requested range at the very end of that. I think we'll benefit more from using prefetched data (by OS) than from doing multiple seeks in this case. I can be wrong, of course. 
Also with search from minSSTableRowId to maxSSTableRowId is there any chance of this becoming a problem?
Do we have any stats on typical data distribution in this case? I mean, it is doable but is it worth it if iteration starts at minSSTableRowId?

Reply:
We don't have any stats. A query can technically request any part of the range. If a query covers the whole range, we have higher level logic that skips this step. I am not familiar enough with OS pre-fetching to know which solution is best.

Reply:
if 
M = maxSSTableRowId - minSSTableRowId
N = number of ordinals in ordinal view

currently we do M * log(N) seeks and read M * log(N) pairs of ints
we will do 1 seek and read N pairs of ints (worst case).
seeks typically worse than sequential read but it's hard to say if extra search for the floor value makes sense without any idea of ranges of M and N and how they overlap.

Reply:
If we assume that `startRowId` / `maxSegmentRowId` is meaningful, maybe we can use that as a naive threshold? It's mostly going to be based on the cost to load some unknown number of buffers, so maybe that won't help.

Reply:
added search for the index.